### ISSUE TITLE
Add support for pack_as_bytes configuration for ext_authz.v3

### ISSUE THREAD

**Issue by sidharthramesh** [knative-sandbox/net-kourier#830]
Hey there!
I've been forwarding an internal ArgoCD through a Knative Service (caddy proxy) to expose it to the internet.
I'm using Kourier along with External Authorization (using OPA)

Today I was trying to execute an ArgoCD CLI command: `argocd cluster add <context>` and it failed with a 403 error, with no logs on OPA. There was no rule defined to disallow this request btw.

After hours of searching through logs, I found that when the Argocd CLI tries to make a grpc web request, and this happens in Kourier:
```
[libprotobuf ERROR external/com_google_protobuf/src/google/protobuf/wire_format_lite.cc:577] String field 'envoy.service.auth.v3.AttributeContext.HttpRequest.body' contains invalid UTF-8 data when serializing a protocol buffer. Use the 'bytes' type if you intend to send raw bytes. 

[2022-04-26T16:34:13.257Z] "POST /cluster.ClusterService/Create HTTP/1.1" 403 UAEX 2239 0 1 - "-" "argocd-client/v2.3.3+07ac038.dirty grpc-go/1.15.0" "cf34b22f-2a40-427a-bdd6-a7c9987e92e3" "argocd.dv.kn.voiceintern.com:443" "-"
```
Basically Envoy is trying to convert every request body into a UTF-8 string. This will not be possible in all cases and will fail on [many edge cases](https://github.com/envoyproxy/envoy/issues/9822).

The good news is that the [`pack_as_bytes` parameter]() enables Envoy to send the request as raw bytes without converting into a UTF-8 string: https://github.com/envoyproxy/envoy/pull/13131

This has also been documented in more detail here: https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto.html?highlight=pack_as_bytes#extensions-filters-http-ext-authz-v3-buffersettings

Proposal: Add a `KOURIER_EXTAUTHZ_PACKASBYTES` environment variable to allow passing the `pack_as_bytes` parameter to Envoy.

**Comment by nak3** [comment]
Thank you for opening this. It seems reasonable for adding the configuration option, do you have any plan how we can add the test (especially e2e test) for this feature?

**Comment by sidharthramesh** [comment]
Hey @nak3, like I mentioned in the PR, I’m very new to Go, Knative and Envoy in general. I don’t really know how to write e2e tests for these.

**Comment by skonto** [comment]
@sidharthramesh hi thanks for raising this, you could do the following:

a) start with fixing the current errors on the PR 
b) add a unit test here: https://github.com/knative-sandbox/net-kourier/blob/main/pkg/config/ext_authz_test.go
c) define an env var such as KOURIER_EXTAUTHZ_PACKASBYTES and make sure it is passed in the config 
by the struct here: https://github.com/knative-sandbox/net-kourier/blob/a72117aa893964da24d92e52c8b6f2f126607f8e/pkg/config/ext_authz.go#L75
d) Then you could define a test run like the one here: https://github.com/knative-sandbox/net-kourier/blob/a72117aa893964da24d92e52c8b6f2f126607f8e/test/e2e-kind.sh#L89-L105 by passing KOURIER_EXTAUTHZ_PACKASBYTES=true.

**Comment by sidharthramesh** [comment]
Hey @skonto thank you for the pointers! I’ll try to take a look and fix the PR this week.
